gradle.properties

```diff
@@ -10,31 +10,178 @@
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
-#Mon Jun 26 09:56:43 AEST 2023
 android.enableJetifier=true
+android.useAndroidX=true
+
+# As of 2024-05-04, v8.4.0 is the latest stable gradle plugin release. See: https://developer.android.com/build/releases/gradle-plugin
+# Previous version (in `dev`: 7.3.1, in this `DevWithUpdatedPackages` branch: 8.3.1)
+# Note: Gradle plugin 8.4.0 requires Gradle version 8.6 - so have updated to that.
+gradlePluginVersion=8.4.0
 
-gradlePluginVersion=7.3.1
 org.gradle.jvmargs=-Xmx2048M -Dkotlin.daemon.jvm.options\="-Xmx2048M"
 org.gradle.unsafe.configuration-cache=true
 
 googleServicesVersion=4.3.12
 kotlinVersion=1.8.21
-android.useAndroidX=true
-appcompatVersion=1.6.1
-coreVersion=1.8.0
-coroutinesVersion=1.6.4
+
+# As of 2024-05-05 v0.0.3 is the latest stable. See: https://mvnrepository.com/artifact/org.signal/aesgcmprovider
+# Previous version: 0.0.3 (stable since May 2019 and no further updates)
+aesgcmProviderVersion=0.0.3
+
+# As of 2024-05-04 v1.9.0 is the latest stable. See: https://mvnrepository.com/artifact/androidx.activity/activity
+# Previous version: 1.5.1
+androidxActivityVersion=1.9.0
+
+# As of 2024-05-04 v1.7.1 is the latest stable release. See: https://mvnrepository.com/artifact/androidx.annotation/annotation
+# Previous version: 1.7.1 (circa Dec 2023 - new alpha/beta/RC builds arriving but nothing yet stable)
+androidxAnnotationVersion=1.7.1
+
+# As of 2024-04-09 v1.6.1 is the latest stable release. See: https://developer.android.com/jetpack/androidx/releases/appcompat
+# Previous version: 1.6.1 (circa Feb 2023 - new alpha & beta builds arriving but nothing yet stable)
+androidxAppcompatVersion=1.6.1
+
+# As of 2024-05-05 v1.0.0 is the latest stable release. See: https://mvnrepository.com/artifact/androidx.cardview/cardview
+# Previous version: 1.0.0 (circa Sept. 2018 - no new updates expected)
+androidxCardViewVersion=1.0.0
+
+# As of 2024-05-05 v1.6.7 is the latest stable release of the Jetpack Compose suite of components.
+# See: https://mvnrepository.com/artifact/androidx.compose.ui/ui
+# See: https://mvnrepository.com/artifact/androidx.compose.ui/ui-tooling
+# See: https://mvnrepository.com/artifact/androidx.compose.runtime/runtime-livedata
+# See: https://mvnrepository.com/artifact/androidx.compose.foundation/foundation-layout
+# See: https://mvnrepository.com/artifact/androidx.compose.material/material
+# Previous versions: 1.5.2 (circa Sept 2023)
+androidxComposeVersion=1.6.7
```

Comment:
Can we go later? e.g. `1.7.0-beta03`

Reply:
I've been avoiding the bleeding edge and sticking with stable releases - is there something in the betas that we really want or would make our lives easier? If so then IMHO that's fine, as long as @ThomasSession is happy with it.

Reply:
I'd almost be tempted because we have a few nice things with 1.7.x but I've been burned using the beta versions in the past where the API changed a couple of times throughout.
If we want to be clean we should use the BOM declaration and let it handle all the versioning. 1.7.0 looks like it'll be ready soon.

Reply:
Current `2024.06.00` BOM has all the 1.6.8 versions, which I think should be okay for now, surely?
See: https://developer.android.com/develop/ui/compose/bom/bom-mapping